tutorials/integration.md

```diff
@@ -345,122 +388,134 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
 
 These are the steps to start the development:
 
-1. Install browserify so it will be available as a script.
-+	
+1. Install browserify so it will be available as a script.	
 ```bash
 sudo npm install -g browserify
 ```
 1. Create and change to a directory.
 1. Run this command to create the initial package definition file:
-+
 ```bash
 npm init -y
 ```
 1. Install the GSN package and its dependencies:
-+
 ```bash
-npm install @opengsn/gsn@"^2.0.0" ethers
+npm install @opengsn/gsn@"^2.0.3" ethers
 ```
 1. Write your code in a file, for example `index.js`. You can use 
 [`require`](https://nodejs.org/en/knowledge/getting-started/what-is-require/) just as you 
 would with Node.js.
 1. To compile the application into browser-compatible JavaScript, use this command:
-+
 ```bash
 browserify index.js -o bundle.js
 ```
-+
 {% hint style="note" %}
-### NOTE:
+##### NOTE:
 At writing there is a bug that causes the output to have some junk characters. 
-{% endhint %}
 Under Linux you can use the `tr` command to solve this:
-+
 ```bash
 browserify index.js | tr -dc '\0-\177' > bundle.js
 ```
+{% endhint %}
 
 
-### The user interface code
+### The user interface code <a id="the_user_interface_code"></a>
 
 [You can see the user interface 
 code here](https://github.com/qbzzt/opengsn/tree/master/01_SimpleUse/ui). Here are the important parts (first in the JavaScript file and then on 
 the HTML page).
 
+&nsbp; 
+
+Import the packages we need:
+
+```javascript
+var Web3 = require( 'web3')
+const gsn = require('@opengsn/gsn')
+const ethers = require("ethers")
+```
+
+&nbsp;
+
 This is the configuration with the addresses of the relevant contracts (on the 
 test network where they are deployed, Kovan) and the maximum acceptable gas price for 
 our messages. The address of the contract we wish to contact is `conf.ourContract`.
 
 ```javascript
 const conf = {
-	ourContract: '0x23Cd0E36bB4727550bc01Cd3A1E8931b6d7CC796',
+	ourContract: '0x10A51A94d096e746E1Aec1027A0F8deCEC43FF63',
 	notOurs:     '0x6969Bc71C8f631f6ECE03CE16FdaBE51ae4d66B1',
-	paymaster:   '0x0572dc46eb6edc950aa37c12fa9c862d4165cbc5',
-	relayhub:    '0x2E0d94754b348D208D64d52d78BcD443aFA9fa52',
-	stakemgr:    '0x0ecf783407C5C80D71CFEa37938C0b60BD255FF8',
+	paymaster:   '0x3f84367c25dC11A7aBE4B9ef97AB78d5D5498bF5',
+	relayhub:    '0xE9dcD2CccEcD77a92BA48933cb626e04214Edb92',
+        forwarder:   '0x0842Ad6B8cb64364761C7c170D0002CC56b1c498',
```

Comment:
I was under impression our new approach is to get these values (`relayHub` and `Forwarder`) form the Paymaster instance, is that correct? @drortirosh

Reply:
indeed. we're no longer configuring client with anything but paymaster.

Reply:
please read relayhub, forwarder from paymaster, and don't hard-code more addresses